src/main/java/com/minecolonies/coremod/util/BlockPosUtil.java

```diff
@@ -441,4 +447,50 @@ public static BlockPos fromEntity(@NotNull final Entity entity)
     {
         return new BlockPos(MathHelper.floor_double(entity.posX), MathHelper.floor_double(entity.posY), MathHelper.floor_double(entity.posZ));
     }
+
+    /**
+     * Calculates the floor level.
+     *
+     * @param position input position.
+     * @param world    the world the position is in.
+     * @return returns BlockPos position with air above.
+     */
+    @NotNull
+    public static BlockPos getFloor(@NotNull BlockPos position, @NotNull final World world)
+    {
+        final BlockPos floor = getFloor(position, 0, world);
+        if (floor == null)
+        {
+            return position;
+        }
+        return floor;
+    }
+
+    /**
+     * Calculates the floor level.
+     *
+     * @param position input position.
+     * @param depth    the iteration depth.
+     * @param world    the world the position is in.
+     * @return returns BlockPos position with air above.
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 50 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)